satpy/readers/hsaf_h5.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019.
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""A reader for HDF5 Snow Cover (SC) file produced by the Hydrology SAF."""
+import logging
+import os
+from datetime import datetime
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+
+class HSAFFileHandler(BaseFileHandler):
+    """File handler for HSAF H5 files."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(HSAFFileHandler, self).__init__(filename,
+                                              filename_info,
+                                              filetype_info)
+
+        self._msg_datasets = {}
+        self._start_time = None
+        self._end_time = None
+        analysis_time = self._get_datetime(filename)
+        self._analysis_time = analysis_time
+
+    @staticmethod
+    def _get_datetime(filename):
+        fname = os.path.basename(filename)
+        dtstr = fname.split('_')[1]
+        return datetime.strptime(dtstr, "%Y%m%d")
+
+    @property
+    def analysis_time(self):
+        """Get validity time of this file."""
+        return self._analysis_time
+
+    def _prepare_variable_for_palette(self, msg, ds_info):
+        colormap = np.array(np.array(msg))
+        return xr.DataArray(colormap, attrs=ds_info, dims=('idx', 'RGB'))
+
+    def get_metadata(self, msg, name):
+        """Get the metadata."""
+        if name == 'SC':
+            ds_info = {
+                'name': 'SC',
+                'filename': self.filename,
+                'data_time': self._analysis_time,
```

Comment:
The time is already parsed from the filename, so no need for additional parsing.
```suggestion
                'data_time': self.start_time,
```